addOns/network/src/main/java/org/zaproxy/addon/network/internal/server/http/handlers/ZapApiHandler.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.network.internal.server.http.handlers;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.Socket;
+import java.util.Objects;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.network.HttpInputStream;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpOutputStream;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.addon.network.server.HttpMessageHandlerContext;
+import org.zaproxy.zap.extension.api.API;
+import org.zaproxy.zap.network.HttpRequestBody;
+
+/** Handles API requests by calling the {@link API}. */
+public class ZapApiHandler extends HttpRequestHandler {
+
+    private static final Logger LOGGER = LogManager.getLogger(ZapApiHandler.class);
+
+    private HandlerState state;
+
+    /**
+     * Constructs a {@code ZapApiHandler} with the given state provider.
+     *
+     * @param state the state provider.
+     * @throws NullPointerException if the given state provider is {@code null}.
+     */
+    public ZapApiHandler(HandlerState state) {
+        this.state = Objects.requireNonNull(state);
+    }
+
+    @Override
+    protected void handleRequest(HttpMessageHandlerContext ctx, HttpMessage msg) {
+        if (!state.isEnabled()) {
+            return;
+        }
+
+        HttpRequestHeader requestHeader = msg.getRequestHeader();
+        if (HttpRequestHeader.CONNECT.equalsIgnoreCase(requestHeader.getMethod())) {
+            return;
+        }
```

Comment:
This seems to be used multiple locations (in this PR and perhaps earlier PRs as well). Would it make sense to extract a utility method?

(Can be a future task, doesn't need to block this.)